etc/ha_healthchecker/ha_healthchecker.py

```diff
@@ -0,0 +1,748 @@
+#!/usr/bin/python3
+import configparser
+import datetime
+import json
+import logging
+import socket
+import subprocess
+import time
+import os
+
+from github import Github
+import iso8601
+from openlabcmd import zk
+import requests
+import six
+
+
+logging.basicConfig(
+    filename="/etc/openlab/ha_healthchecker/ha_healthchecker.log",
+    filemode='a',
+    format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+    datefmt='%H:%M:%S',
+    level=logging.DEBUG)
+LOG = logging.getLogger(__name__)
+
+
+# Post user info
+{% if use_test_account %}
+ISSUE_USER_TOKEN = "{{ test_github_token }}"
+REPO_NAME = "{{ test_repo_name }}"
+{% else %}
+ISSUE_USER_TOKEN = "{{ github_token }}"
+REPO_NAME = 'theopenlab/openlab'
+{% endif %}
+
+# Simpledns provider
+base_api_url = 'https://api.dnsimple.com/v2/'
+DOMAIN_NAME = 'openlabtesting.org'
+{% if use_test_url %}
+DOMAIN_NAME1 = 'test-status.openlabtesting.org'
+DOMAIN_NAME2 = 'test-logs.openlabtesting.org'
+DOMAIN_NAME_BK = 'test-logs-bak.openlabtesting.org'
+{% else %}
+DOMAIN_NAME1 = 'status.openlabtesting.org'
+DOMAIN_NAME2 = 'logs.openlabtesting.org'
+DOMAIN_NAME_BK = 'logs-bak.openlabtesting.org'
+{% endif %}
+TOKEN = "{{ dns_access_token }}"
+ACCOUNT_ID = {{ dns_account_id }}
+TARGET_ORI_IP = "{{ ori_master_ip }}"
+TARGET_ORI_BK_IP = "{{ ori_backup_ip }}"
+TARGET_CHANGE_IP = "{{ slave_ip }}"
+TARGET_CHANGE_BK_IP = "{{ slave_backup_ip }}"
+
+# General constants
+SYSTEMCTL_STATUS = 'status'
+SYSTEMCTL_RESTART = 'restart'
+SYSTEMCTL_STOP = 'stop'
+SYSTEMCTL_START = 'start'
+SYSTEMCTL_ENABLE = 'enable'
+SYSTEMCTL_DISABLE = 'disable'
+
+
+class Base(object):
+    def __init__(self, zk, heartbeat_timeout):
+        self.zk = zk
+        self.heartbeat_timeout = heartbeat_timeout
+        self.node = self.zk.get_node(socket.gethostname())
+        self.oppo_node = self._get_oppo_node()
+
+    def _get_oppo_node(self):
+        for zk_node in self.zk.list_nodes():
+            # During swtich, we can not get oppo node
+            if (zk_node.role != self.node.role and
```

Comment:
It can be just: ` if zk_node.type == self.node.type and zk_node.name != self.node.name:`